influx-reports/influx.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+import argparse
+import datetime
+import dateutil.parser
+import logging
+import os
+import sys
+import yaml
+from collections import defaultdict
+from influxdb import InfluxDBClient
+from subunit.v2 import StreamResultToBytes
+
+
+def return_time(client, query, delta_seconds=0):
+    """ From an InfluxDB query, fetch
+    the first point time, and return a
+    python time object. Shift it from
+    a few seconds (delta_seconds) if necessary.
+    """
+    timestamp_query = client.query(query)
+    # Get points is generator, we should just get first
+    # point time (string type).
+    fpt_str = next(timestamp_query.get_points())['time']
+    try:
+        fpt = dateutil.parser.parse(fpt_str) + datetime.timedelta(seconds=delta_seconds)
+    except Exception as e:
+        raise SystemExit(e)
```

Comment:
Why escalate the exception to SystemExit, when an uncaught ValueError or TypeError will halt execution?